code/code/task/task_picklock.cc

```diff
@@ -1,144 +1,122 @@
     ch->stopTask();
-    if (this != ch->unequip(ch->getPrimaryHold())) {
-      vlogf(LOG_BUG, "whacked out unequip in task_picklock");
-      return;
-    }
-    delete this;
     return;
   }
-  difficulties = 3 * exitp->lock_difficulty;
-  if (exitp->lock_difficulty >= 100) {
-    // guaranteed unpickable, see if they recognize this fact 
-    if (ch->bSuccess(skill, SKILL_PICK_LOCK)) {
-      act("This lock is totally impossible to pick.  You give up.",
-          TRUE, ch, this, 0, TO_CHAR);
-      ch->stopTask();
-      return;
-    }
+
+  int difficulty = 3 * exit->lock_difficulty;
+  if ((!pickable || difficulty > skill) &&
+      difficulty > (skill + 100) &&
+      !ch->isDextrous()) {
+    act("Uhoh, $n seems to have jammed the lock!", true, ch, nullptr, nullptr, TO_ROOM);
+    ch->sendTo("Uhoh.  You seemed to have jammed the lock!\n\r");
+    SET_BIT(exit->condition, EXIT_JAMMED);
+    ch->stopTask();
+    return;
   }
-  if ((exitp->lock_difficulty >= 100) || (difficulties > skill)) {
-    if ((difficulties > (skill + 100)) && !ch->isDextrous()) {
-      act("Uhoh, $n seems to have jammed the lock!",TRUE,ch,0,0,TO_ROOM);
-      ch->sendTo("Uhoh.  You seemed to have jammed the lock!\n\r");
-      exitp->lock_difficulty = min(100,exitp->lock_difficulty+10);
-      ch->stopTask();
-      return;
-    } else if (!::number(0,2)) {
-      act("You wiggle $p in the lock, without luck. (yet...)",
-          TRUE,ch,this,0,TO_CHAR);
-    }
-  } else if (ch->bSuccess(skill, SKILL_PICK_LOCK)) {
-    // this used to just be an automatic for the else case
-    // but what the heck, this makes it a bit harder to pick
-    // and helps make pick-lock a learn-by-doing too
-
-    REMOVE_BIT(exitp->condition, EXIT_LOCKED);
-    if (!exitp->keyword.empty()) {
-      act(format("$n skillfully picks the lock of the %s.\n\r") % 
-	  fname(exitp->keyword),
-	  TRUE,ch,0,0,TO_ROOM);
-    } else
-      act("$n picks the lock.", TRUE, ch, 0, 0, TO_ROOM);
+
+  if (pickable && ch->bSuccess(skill, SKILL_PICK_LOCK)) {
+    REMOVE_BIT(exit->condition, EXIT_LOCKED);
+    sstring msg = !exit->keyword.empty() ?
+        sstring(format("$n skillfully picks the lock of the %s.") % fname(exit->keyword)) :
+        sstring("$n picks the lock.");
+    act(msg, true, ch, nullptr, nullptr, TO_ROOM);
     ch->sendTo("The lock quickly yields to your skills.\n\r");
 
     // now for unlocking the other side, too
-    temp_rp = real_roomp(exitp->to_room);
-    if (temp_rp &&
-
-        (back = temp_rp->dir_option[rev_dir[ch->task->status]]) &&
+    TRoom *other = real_roomp(exit->to_room);
+    roomDirData *back;
+    if (other &&
+        (back=other->dir_option[rev_dir[ch->task->status]]) &&
         back->to_room == ch->in_room)
       REMOVE_BIT(back->condition, EXIT_LOCKED);
     ch->stopTask();
+    return;
   }
+
+  if (!::number(0,2))
+    act("You wiggle $p in the lock, without luck. (yet...)", true, ch, tool, nullptr, TO_CHAR);
```

Comment:
Me no speaking English natively, but perhaps "the lockpick"?

Reply:
I agree that with $p printing short_desc it's not strictly grammatical in this sentence, since most of the tools are plural, but I feel printing short_desc is the "sneeziest" thing to do here and consistent with much of our other act usage. With "a skeleton key" in the database, printing a specific noun here can mismatch with the held tool.